documentation/bolt_examples.md

```diff
@@ -123,3 +123,223 @@ To convert the `restart_service.ps1` script to a task, giving you the ability to
+
+2. Create an inventory file to store information about your nodes. This is stored as `inventory.yaml` by default in the project directory. Add the following code: 
+
+```
+groups:
+  - name: windows
+    nodes:
+      - chocowin0.classroom.puppet.com
+      - chocowin1.classroom.puppet.com
+    config:
+      transport: winrm
+      winrm:
+        user: Administrator
+        password: <ADD PASSWORD>
+```
+
+4. To make sure that your inventory is configured correctly and that you can connect to all the target nodes, run the following command from inside the project directory: 
+
+```
+bolt command run 'echo hi' --targets windows
+```
+
+**Note:** The `--targets windows` argument refers to the node group defined in the inventory file.
+
+You should get the following output:
+
+```
+Started on x.x.x.x...
+Started on localhost...
+Finished on localhost:
+  STDOUT:
+    hi
+Finished on 0.0.0.0:
+  STDOUT:
+    hi
+Finished on 127.0.0.1:
+  STDOUT:
+    hi
+Successful on 3 nodes: 0.0.0.0:20022,localhost
+Ran on 3 nodes in 0.20 seconds
+```
+
+### 2. Download the Chocolatey module and its dependencies from the Puppet Forge. 
+
+Bolt uses a `Puppetfile` to install module content from the Forge. A `Puppetfile` is a formatted text file that specifies the modules and data you want in each environment.
+
+1 . Create a file named `Puppetfile` in the project directory, with the modules needed for this example:
+
+```     
+# Modules from the Puppet Forge
+mod 'puppetlabs-chocolatey', '4.1.0'
+## dependencies
+mod 'puppetlabs-stdlib', '4.13.1' #install latest
+mod 'puppetlabs-powershell', '2.3.0'
+mod 'puppetlabs-registry', '2.1.0'
+        
+# Modules from Git
+# mod 'apache',
+# :git    => 'https://github.com/puppetlabs/puppetlabs-apache',
+# :commit => 'de290646f97e04b4b8e42c70f6e01e860c394ce7'     
+ ```
+
+Note that you can install modules from a number of different sources. For more information, see the [Puppetfile readme](https://github.com/puppetlabs/r10k/blob/master/doc/puppetfile.mkd#examples).
+
+2. Add the Puppetfile to a remote repository, for example, GitHub.
+
+3. Add the following code to the Puppetfile — with your git user account: 
+
+```     
+mod 'puppet_choco_tap',
+    :git    =>  'https://github.com/<your git user>/puppet_choco_tap.git'
+```
+
+4. From inside the project directory, install the required modules:
+
+```
+bolt puppetfile install
+```
+
+After it runs, you can see a `modules` directory inside the project directory, containing the modules you specified in the `Puppetfile`.
+
+### 3. Write a Bolt plan to apply Puppet code and orchestrate the deployment of a package resource using the Chocolatey provider. Plans allow you to run more than one task with a single command.
+
+1. Create a `site-modules` directory. This is where you will add code and local modules.
```

Comment:
@adreyer What do you mean by local modules?

Reply:
modules that are committed directly to the project repository instead of managed by the Pupeptfile as seperate git repos or forge modules